lib/index.js

```diff
@@ -353,8 +329,15 @@ class KuzzleCluster {
     this.node.state.locks.create.add(payload.roomId);
   }
 
-  roomDeleted (roomId) {
+  async removeRoom (roomId) {
     this.node.state.locks.delete.add(roomId);
+    const room = this._rooms.flat.get(roomId);
+
+    if (room && room.count > 1) {
+      throw new Error('cannot remove: still used by other nodes');
```

Comment:
Shouldn't this error be an `InternalError` ?

Reply:
It doesn't matter, it's only used to interrupt the pipe call. Kuzzle will catch this error and discard it anyway (see https://github.com/kuzzleio/kuzzle/blob/loose-coupling-realtime-redux/lib/core/realtime/hotelClerk.js#L591)

It's dirty. Hopefully with the new cluster we won't need to resort to such tricks: this mechanism will be removed altogether. With the new cluster architecture, nodes won't need to maintain rooms they do not use.